js/components/utils.js

```diff
@@ -465,6 +466,29 @@ var Utils = {
     return destination;
   },
 
+  /**
+   * Checks if image loads and is valid
+   *
+   * @function isValidImage
+   * @param {String} src - absolute path to image src
+   * @returns {Boolean} true if image is valid, false if not
+   */
+  isValidImage: function(src) {
+    if (src) {
```

Comment:
Would it be better here to check that `src` is a string and also not empty? Anything of truthy value will pass here (i.e. 1, a non-empty object, etc)